### PR TITLE
Add socketfilterfw parser table

### PR DESCRIPTION
**Pull request by Micah-Kolide**
With the release of MacOS 15, Apple has deprecated the use of a plist for application firewall settings. Instead they say to use `socketfilterfw` if you relied on the plist in past versions.

osquery does not have a pattern for executing and parsing data from binaries (yet), so I've added this table to fill in the gap if that table stays unsupported for MacOS 15.

### CODE REVIEW

**Comment by Micah-Kolide** [comment]
Will be adding some tests next, but as for an example query:
```
osquery> SELECT * FROM kolide_socketfilterfw;
+------------------------+--------+----------------------+-----------+-------+
| fullkey                | parent | key                  | value     | query |
+------------------------+--------+----------------------+-----------+-------+
| 0/logging_option       | 0      | logging_option       | throttled | *     |
| 0/stealth_enabled      | 0      | stealth_enabled      | false     | *     |
| 0/allow_signed_enabled | 0      | allow_signed_enabled | true      | *     |
| 0/block_all_enabled    | 0      | block_all_enabled    | false     | *     |
| 0/global_state         | 0      | global_state         | true      | *     |
| 0/logging_enabled      | 0      | logging_enabled      | true      | *     |
+------------------------+--------+----------------------+-----------+-------+
```

File: ee/tables/execparsers/socketfilterfw/parser.go
Comment (directionless) [inline ee/tables/execparsers/socketfilterfw/parser.go:20]: I'd recommend putting `row` into the `for scanner.Scan()` loop.

Outside the loop it optimizes reducing memory allocations, which can be a small win for high performance, but comes with the risk/overhead of needing to make sure you've cleared state each time. Since we're not high performance, it's simpler to put it into the loop and let the scoping handle it.

File: ee/tables/execparsers/socketfilterfw/parser.go
Comment (directionless) [inline ee/tables/execparsers/socketfilterfw/parser.go:47]: This is an interesting questions. Do you think this format is stable enough we should use the enum here, or should we capture everything and try to coerce it to kv pairs?

File: ee/tables/execparsers/socketfilterfw/parser.go
Comment (directionless) [inline ee/tables/execparsers/socketfilterfw/parser.go:51]: When would this occur?

**Review by RebeccaMahany — COMMENTED** [review]
LGTM once seph's comments are addressed!

File: ee/tables/execparsers/socketfilterfw/parser.go
Comment (Micah-Kolide) [reply ee/tables/execparsers/socketfilterfw/parser.go:51]: I had placed this in for the malformed test data as the `state` string for global state is matched in the app states. Now that data shouldn't be ever received, so this may be overkill.

File: ee/tables/execparsers/socketfilterfw/parser.go
Comment (Micah-Kolide) [reply ee/tables/execparsers/socketfilterfw/parser.go:20]: Sounds good

Comment (Micah-Kolide) [reply ee/tables/execparsers/socketfilterfw/parser.go:23]: I updated how row is being handled since I added in apps, but I still have it first declared outside of the loop.

File: ee/tables/execparsers/socketfilterfw/parser.go
Comment (Micah-Kolide) [reply ee/tables/execparsers/socketfilterfw/parser.go]: I've made use of this space for the apps table.

File: ee/tables/execparsers/socketfilterfw/parser.go
Comment (Micah-Kolide) [reply ee/tables/execparsers/socketfilterfw/parser.go]: Resolved

File: ee/tables/execparsers/socketfilterfw/parser.go
Comment (Micah-Kolide) [reply ee/tables/execparsers/socketfilterfw/parser.go:24]: Haha yeah of course I don't know why I randomly switched in the middle of making this. snake case always is stuck in my head 😄

File: ee/tables/execparsers/socketfilterfw/parser.go
Comment (Micah-Kolide) [reply ee/tables/execparsers/socketfilterfw/parser.go:37]: I probably didn't need to include it, but yeah I should've done it below as well if I was to keep this pattern.

File: ee/tables/execparsers/socketfilterfw/parser.go
Comment (Micah-Kolide) [reply ee/tables/execparsers/socketfilterfw/parser.go:45]: I lower cased the line parsing since I'm creating the key string manually, but I didn't want to change the case of app names, so I kept it off there.

File: ee/tables/execparsers/socketfilterfw/parser.go
Comment (Micah-Kolide) [reply ee/tables/execparsers/socketfilterfw/parser.go:20]: Thank you :)

Now I just need to do the same for the rest of my PRs 😅

File: ee/tables/execparsers/socketfilterfw/parser.go
Comment (Micah-Kolide) [reply ee/tables/execparsers/socketfilterfw/parser.go:89]: I agree and went a bit back and forth on it until I landed here. I'm not opposed to splitting them out, and I think it does make it more readable and easier to track.

I can see a couple of differences I'll need to do from your example, but I'll see how I can re-work it to be closer to that pattern.

File: ee/tables/execparsers/socketfilterfw/parser.go
Comment (RebeccaMahany) [inline ee/tables/execparsers/socketfilterfw/parser.go:44]: I think you could combine like this -- it would let you get rid of `parseAppData` bool entirely.

```suggestion
		if strings.Contains(line, "Total number of apps") {
			if appRow := parseAppList(line); appRow != nil {
				results = append(results, appRow)
			}
			continue
		}
```

File: ee/tables/execparsers/socketfilterfw/parser.go
Comment (RebeccaMahany) [inline ee/tables/execparsers/socketfilterfw/parser.go:49]: Ahh, I understand better what `row` looks like now

**Review by RebeccaMahany — APPROVED** [review]
LGTM, had a final nitpick for conciseness

File: ee/tables/execparsers/socketfilterfw/parser.go
Comment (Micah-Kolide) [reply ee/tables/execparsers/socketfilterfw/parser.go:44]: I'm confused how this is supposed to work. Only the first line of text for the app list contains that string:
```
Total number of apps = 6
replicatord (state: 1)
Pop Helper.app (state: 4)
Google Chrome (state: 4)
rtadvd (state: 1)
com.docker.backend (state: 1)
sshd-keygen-wrapper (state: 1)
```

Comment (RebeccaMahany) [reply ee/tables/execparsers/socketfilterfw/parser.go:44]: Oh blah, I'm sorry, I see what I missed! Ignore this.

**Review by RebeccaMahany — APPROVED** [review]

File: ee/tables/execparsers/socketfilterfw/parser.go
Comment (directionless) [reply ee/tables/execparsers/socketfilterfw/parser.go]: Working across all these languages with different idioms always throws me

File: ee/tables/execparsers/socketfilterfw/parser.go
Comment (directionless) [reply ee/tables/execparsers/socketfilterfw/parser.go:23]: Oh, I see now. 

In the general case, you're returning a single row, with a lot of columns. 

I was thinking you could return this as EAV style data, and let data flatten handle it.

File: ee/tables/execparsers/socketfilterfw/parser.go
Comment (Micah-Kolide) [reply ee/tables/execparsers/socketfilterfw/parser.go:44]: No worries, I just wasn't sure if I was missing something

**Pull request merged**
